### DISCUSSION TITLE
Issues Installing

### DISCUSSION THREAD

**Discussion by JohnStratford** [cnr-isti-vclab/TagLab#118] [General]
Hi,

I'd love to use the software, but I'm having trouble installing it. I have been following the 'Installing TagLab' wiki (https://github.com/cnr-isti-vclab/TagLab/wiki/Install-TagLab) but have had issues at a couple of Steps and I'm wondering if anyone is able to point out where I am going wrong, please..

I am using Windows 11, and believe I have installed 64-bit Python 3.10.11, CUDA Toolkit 11.6 and the x64 Visual Studio Redistributable. When installing Python I selected the box to add it to the Path and, when provided with the option, I clicked to Disable PATH Limit Length (when I first downloaded Python I did this; after I had issues I uninstalled Python then re-installed it - when I re-installed it, I was never given the option to Disable PATH Limit Length). I have also downloaded and unzipped the 'TagLab-main' folder found at the 'Download Zip' button under the 'Code' tab on the GitHub page. The shell I have used is Windows PowerShell.
![Software installed + file downloaded](https://github.com/cnr-isti-vclab/TagLab/assets/146823049/94f7a515-696a-4bec-99ab-bbbba060904f)

**Step 0:**
The first issue I have is that "The term 'python3' is not recognized" when I 'check if Python and CUDA are properly installed'. CUDA appears to be installed ok. And when I input 'python' rather than 'python3', it is able to return the correct version: 

 PS C:\Users\johns> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Fri_Dec_17_18:28:54_Pacific_Standard_Time_2021
Cuda compilation tools, release 11.6, V11.6.55
Build cuda_11.6.r11.6/compiler.30794723_0
PS C:\Users\johns> python3 --version
python3 : The term 'python3' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ python3 --version
+ ~~~~~~~
+ CategoryInfo          : ObjectNotFound: (python3:String) [], CommandNotFoundException
+ FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\johns> python --version
Python 3.10.11

Do you know where I have gone wrong, and how I fix my error?

**Step 2:**
Despite that error, I continued to see if it would work. When I run Step 2, I am presented with "ModuleNotFoundError: No module named 'osgeo'":

[notice] A new release of pip is available: 23.0.1 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
Traceback (most recent call last):
  File "C:\Users\johns\Downloads\TagLab + required packages\TagLab-main\install.py", line 275, in <module>
    gdal_is_installed = importutil.find_spec("osgeo.gdal")
  File "C:\Users\johns\AppData\Local\Programs\Python\Python310\lib\importlib\util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
ModuleNotFoundError: No module named 'osgeo'

Do you know why this error is occurring and how I can fix it?
A copy of my full script: [Full script with errors.txt](https://github.com/cnr-isti-vclab/TagLab/files/12801388/Full.script.with.errors.txt)

Apologies for the hassle, but thank you very much for your help!
John

**Comment by Jordan-Pierce** [comment]
@JohnStratford You're running into the issue in `install.py` where it's trying to install `rasterio` and `gdal`, the latter is sometime problematic to install in general. For some reason, the wheel's TagLab tries to install doesn't work for me either, but if you just go to the original source, you can download them, install them using `pip` before running `install.py` again, and that section will be skipped since you already have it installed. 

[Here's the link the archive](https://www.lfd.uci.edu/~gohlke/pythonlibs/#gdal), you'll need whatever version based on your python version (3.10), so it looks like probably `GDAL‑3.4.3‑cp310‑cp310‑win_amd64.whl`; download the most recent version of `rasterio` as well. Go to your download folder, and install these in your conda environment using `pip`, then re-run `install.py`

**Reply by JohnStratford** [reply]
Thanks @Jordan-Pierce ! Installing GDAL separately worked and I'm in! Thanks for your help
